snippets/Google.Storage.V1.Snippets/StorageClientSnippets.cs

```diff
@@ -12,33 +12,212 @@
-            var objects = client.ListObjects(bucketId, "Hello");
+
+            // List only objects with a name starting with "greet"
+            var objects = client.ListObjects(bucketName, "greet");
             // </ListObjects>
+
             var names = objects.Select(obj => obj.Name).ToList();
-            Assert.Contains(_fixture.HelloWorldName, names);
+            Assert.Contains(fixture.HelloWorldName, names);
         }
+
+        [Fact]
+        void DownloadFile()
+        {
+            var bucketName = fixture.BucketName;
+            var projectId = fixture.ProjectId;
+
+            // <DownloadObject>
+            var client = StorageClient.Create();
+            var source = "greetings/hello.txt";
+            var destination = @".\hello.txt";
+
+            using (var stream = File.Create(destination))
+            {
+                // IDownloadProgress defined in Google.Apis.Download namespace
+                var progress = new Progress<IDownloadProgress>(
+                    p => Console.WriteLine($"bytes: {p.BytesDownloaded}, status: {p.Status}")
+                );
+
+                // Download source object from bucket to local file system
+                client.DownloadObject(bucketName, source, stream, null, progress);
+            }
+            // </DownloadObject>
+
+            // want to show the source path in the snippet to give an idea
+            // of nested paths, but want to make sure that the pre-uploaded file
+            // is at the same path as the source since it is duplicated data
+            Assert.Equal(fixture.HelloWorldName, source);
+
+            Assert.Equal(fixture.HelloWorldContent, File.ReadAllText(destination));
+            File.Delete(destination);
+        }
+
+        [Fact]
+        void UploadFile()
+        {
+            // need to write this at test run instead of fixture creation
+            // as the DownloadFile test creates and deletes the same file
+            var local = @".\hello.txt";
+            File.WriteAllText(local, fixture.HelloWorldContent);
+            var bucketName = fixture.BucketName;
+
+            // <UploadObject>
+            var client = StorageClient.Create();
+            var source = @".\hello.txt";
+            var destination = "greetings/hello.txt";
+            var contentType = "text/plain";
+
+            using (var stream = File.OpenRead(source))
+            {
+                // IUploadProgress defined in Google.Apis.Upload namespace
+                var progress = new Progress<IUploadProgress>(
+                  p => Console.WriteLine($"bytes: {p.BytesSent}, status: {p.Status}")
+                );
+
+                // ObjectsResource defined in Google.Apis.Storage.v1 namespace
+                // var acl = ObjectsResource.InsertMediaUpload.PredefinedAclEnum.PublicRead // public
+                var acl = ObjectsResource.InsertMediaUpload.PredefinedAclEnum.AuthenticatedRead; // private
```

Comment:
That would certainly seem like a reasonable output format, but will require more work to implement. (The current extraction is almost entirely from docfx - I just automated attaching the snippets to the prod methods.)

If we decide to divorce ourself from the automated extraction of course, that does mean we could do more cunning processing - like "inlining" constants to avoid the tension between code duplication and being standalone.